src/main.rs

```diff
@@ -39,13 +43,23 @@ fn main() {
         })
         .insert_resource(ClearColor(Color::BLACK))
         .add_plugins(DefaultPlugins)
+        .init_resource::<MenuMaterials>()
         // AppState
-        .insert_resource(State::new(AppState::Overworld))
+        .insert_resource(State::new(AppState::MainMenu))
         .add_stage_before(stage::UPDATE, APPSTATES, StateStage::<AppState>::default())
+        // Main menu
+        .on_state_enter(APPSTATES, AppState::MainMenu, menu::main_menu::setup.system())
+        .on_state_update(APPSTATES, AppState::MainMenu, menu::button_interact.system())
+        .on_state_exit(APPSTATES, AppState::MainMenu, despawn_all::<menu::StateCleanup>.system())
+        // Settings menu
+        .on_state_enter(APPSTATES, AppState::SettingsMenu, menu::settings::setup.system())
+        .on_state_update(APPSTATES, AppState::SettingsMenu, menu::button_interact.system())
+        .on_state_exit(APPSTATES, AppState::SettingsMenu, despawn_all::<menu::StateCleanup>.system())
```

Comment:
I think it is better if you don't reuse the same marker type for main menu and settings menu entities.

It will be less prone to breakage if we want to change the flow in the future.

Reply:
Isn't it a different type, just named the same? They're from different modules.

Reply:
> Isn't it a different type, just named the same? They're from different modules.

It's a different one from the game one, but I did reuse it for the main menu & settings menu. I've now fixed this and used separate marker components for both.

Reply:
Umm, no? `menu::StateCleanup` and `overworld::StateCleanup` are separate types. But it seems to me that both menus are reusing the same `menu::StateCleanup`.

Reply:
Right, I missed that part. It is indeed fixed now.